src/TestFramework/TestFramework/Assertions/CollectionAssert.cs

```diff
@@ -302,45 +510,110 @@ public static void AllItemsAreUnique([NotNull] ICollection? collection, string?
 
-                    Assert.ThrowAssertFailed("CollectionAssert.AllItemsAreUnique", finalMessage);
-                }
-            }
-            else
-            {
-#pragma warning disable CA1864 // Prefer the 'IDictionary.TryAdd(TKey, TValue)' method
-                if (table.ContainsKey(current))
-                {
-                    string userMessage = Assert.BuildUserMessage(message, parameters);
-                    string finalMessage = string.Format(
-                        CultureInfo.CurrentCulture,
-                        FrameworkMessages.AllItemsAreUniqueFailMsg,
-                        userMessage,
-                        Assert.ReplaceNulls(current));
+    /// <summary>
+    /// Tests whether all items in the specified collection are unique or not and
+    /// throws if any two elements in the collection are equal.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of values to compare.
+    /// </typeparam>
+    /// <param name="collection">
+    /// The collection in which to search for duplicate elements.
+    /// </param>
+    /// <param name="message">
+    /// The message to include in the exception when <paramref name="collection"/>
+    /// contains at least one duplicate element. The message is shown in
+    /// test results.
+    /// </param>
+    /// <exception cref="AssertFailedException">
+    /// <paramref name="collection"/> is null, or <paramref name="collection"/> contains at least one duplicate
+    /// element.
+    /// </exception>
+    public static void AllItemsAreUnique<T>([NotNull] IEnumerable<T?>? collection, string? message)
+        => AllItemsAreUnique(collection, message, null);
+
+    /// <summary>
+    /// Tests whether all items in the specified collection are unique or not and
+    /// throws if any two elements in the collection are equal.
+    /// </summary>
+    /// <typeparam name="T">
+    /// The type of values to compare.
+    /// </typeparam>
+    /// <param name="collection">
+    /// The collection in which to search for duplicate elements.
+    /// </param>
+    /// <param name="message">
+    /// The message to include in the exception when <paramref name="collection"/>
+    /// contains at least one duplicate element. The message is shown in
+    /// test results.
+    /// </param>
+    /// <param name="parameters">
+    /// An array of parameters to use when formatting <paramref name="message"/>.
+    /// </param>
+    /// <exception cref="AssertFailedException">
+    /// <paramref name="collection"/> is null, or <paramref name="collection"/> contains at least one duplicate
+    /// element.
+    /// </exception>
+    public static void AllItemsAreUnique<T>([NotNull] IEnumerable<T?>? collection, [StringSyntax(StringSyntaxAttribute.CompositeFormat)] string? message,
+        params object?[]? parameters)
+    {
+        Assert.CheckParameterNotNull(collection, "CollectionAssert.AllItemsAreUnique", "collection", string.Empty);
```

Comment:
Please do call out if the method was only moved or if the content was changed, that's easing reviewer's job.